File: androidApp/src/main/java/com/mbta/tid/mbta_app/android/component/legacyRouteCard/LegacyRouteCard.kt
package com.mbta.tid.mbta_app.android.component.legacyRouteCard

import androidx.compose.runtime.Composable
import com.mbta.tid.mbta_app.android.component.PinButton
import com.mbta.tid.mbta_app.model.Route

@Composable
fun LegacyRouteCard(
    route: Route,
    pinned: Boolean,
    onPin: (String) -> Unit,
    content: @Composable () -> Unit
) {
    TransitCard(
        header = {
            RouteHeader(route) { textColor ->
                PinButton(pinned = pinned, color = textColor) { onPin(route.id) }
            }
        },
        content = content
    )
}

File: iosApp/iosApp/ComponentViews/LegacyRouteCard/LegacyRouteCard.swift
//
//  LegacyRouteCard.swift
//  iosApp
//
//  Created by Brandon Rodriguez on 4/16/24.
//  Copyright © 2024 MBTA. All rights reserved.
//

import Shared
import SwiftUI

struct LegacyRouteCard<Content: View>: View {
    let route: Route
    let pinned: Bool
    let onPin: (String) -> Void
    let content: () -> Content

    @ScaledMetric private var modeIconHeight: CGFloat = 24

    var body: some View {
        TransitCard(header: {
            RouteHeader(route: route) {
                PinButton(pinned: pinned, action: { onPin(route.id) })
            }.accessibilityAddTraits(.isButton)
        }, content: content)
    }
}
